tasks/components/layout/OrganizationDetails.tsx

```diff
@@ -101,23 +72,35 @@ export const OrganizationDetail = ({
         onBackgroundClick={() => setIsShowingConfirmDialog(false)}
         onConfirm={() => {
           setIsShowingConfirmDialog(false)
-          onDelete(newOrganization)
+          onDelete(organizationForm.organization)
         }}
         confirmType="negative"
       />
       <ColumnTitle title={translation.organizationDetail}/>
       <OrganizationForm
-        organization={newOrganization}
+        organization={organizationForm.organization}
         onChange={(newOrganizationDetails, isValid) => {
-          setNewOrganization({ ...newOrganization, ...newOrganizationDetails })
-          setFilledRequired(isValid)
+          setOrganization({
+            hasChanges: true,
+            isValid,
+            organization: { ...organizationForm.organization, ...newOrganizationDetails },
+          })
         }}
         isShowingErrorsDirectly={!isCreatingNewOrganization}
       />
       <div className={tw('mt-6')}>
         <OrganizationMemberList
-          members={newOrganization.members}
-          onChange={(members) => setNewOrganization({ ...newOrganization, members })}
+          members={organizationForm.organization.members}
+          onChange={(members) => setOrganization((prevState) => {
+            return {
+              hasChanges: true,
+              isValid: prevState.isValid,
+              organization: {
+                ...prevState.organization,
+                members
+              }
+            }
```

Comment:
Btw there is a trick for this. You can do `() => ({key: "value"})`

Reply:
in general yes, but not in this case because the previousStates organization is required, or how would you like to resolve this?

Reply:
```js
onChange={(members) => setOrganization((prevState) => ({ isValid: prevState.isValid, test: 'test' }))}
```